src/components/mypage/Info.tsx

```diff
@@ -1,14 +1,192 @@
+import { myInfoSchema } from '@/libs/utils/schemas/myInfoSchema';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { useRouter } from 'next/navigation';
+import React, { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
 
 import Button from '../commons/Button';
+import Input from '../commons/Input/Input';
+import Modal from '../commons/Modal';
+
+type FormData = {
+  nickname: string;
+  email: string;
+  newPassword?: string | null;
+  reEnterPassword?: string | null;
+};
+
+type registerDataType = {
+  nickname?: string;
+  newPassword?: string;
+};
+
+interface ApiError extends Error {
+  response?: {
+    data?: {
+      message?: string;
+    };
+  };
+}
+
+const fetchUserInfo = async () => {
+  const userInfo = await getUser();
+  return userInfo;
+};
 
 export default function Info() {
+  const { data: userInfo } = useQuery({
+    queryKey: ['userInfo'],
+    queryFn: fetchUserInfo,
+  });
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    trigger,
+    reset,
+  } = useForm<FormData>({
+    resolver: yupResolver(myInfoSchema),
+    mode: 'all',
+  });
+
+  const [modalMessage, setModalMessage] = useState('');
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isSuccessMessage, setIsSuccessMessage] = useState(false);
+
+  const router = useRouter();
+
+  useEffect(() => {
+    if (userInfo) {
+      reset({
+        nickname: userInfo.nickname,
+        email: userInfo.email,
+        newPassword: '',
+        reEnterPassword: '',
+      });
+    }
+  }, [userInfo, reset]);
+
+  const mutation = useMutation({
+    mutationFn: patchUserInfo,
+    onSuccess: () => {
+      console.log('정보 수정 성공');
+      setModalMessage('수정 완료!');
+      setIsModalOpen(true);
+      setIsSuccessMessage(true);
+    },
+    onError: (error: ApiError) => {
+      if (error.response && error.response.data) {
+        if (error.response.data.message === '수정할 내용이 없습니다.') {
+          setModalMessage('변경된 정보가 없습니다.');
+        } else if (
+          error.response.data.message === '닉네임은 10자 이하로 작성해주세요.'
+        ) {
+          setModalMessage('닉네임은 10자 이하로 작성해주세요.');
+        } else {
+          console.error('로그인 실패', error);
+        }
+        setIsModalOpen(true);
+        setIsSuccessMessage(false);
+      }
+    },
+  });
+
+  const onSubmit = async (data: FormData) => {
+    const { nickname, newPassword } = data;
+
+    // 유효성 검사: 최소 하나의 필드가 유효한 경우
+    if (!nickname && !newPassword) {
+      console.log('변경된 정보가 없습니다.');
+      return;
+    }
+
+    const registerData: registerDataType = {
+      nickname: nickname !== userInfo?.nickname ? nickname : undefined,
+      newPassword: newPassword?.trim() || undefined,
+    };
+
+    // newPassword가 빈 문자열이 아닐 경우에만 추가
+    if (newPassword?.trim()) {
+      registerData.newPassword = newPassword.trim();
+    }
+
+    mutation.mutate(registerData);
+  };
```

Comment:
```suggestion
    const registerData: registerDataType = {
      nickname: nickname !== userInfo?.nickname ? nickname : undefined,
      newPassword: newPassword?.trim() || undefined,
    };

    mutation.mutate(registerData);
  };
```
`newPassword` 할당해주는 부분이 중복되는 것 같은데 확인 한번만 해주시면 좋을 것 같습니다!

Reply:
스키마에 공백 처리하는 로직 추가했습니다!! 저부분 없어도 작동 잘 되네요 👍🏻🥰